src/extent.c

```diff
@@ -559,23 +551,26 @@ extent_recycle(tsdn_t *tsdn, pac_t *pac, ehooks_t *ehooks, ecache_t *ecache,
     bool *commit, bool growing_retained) {
 	witness_assert_depth_to_rank(tsdn_witness_tsdp_get(tsdn),
 	    WITNESS_RANK_CORE, growing_retained ? 1 : 0);
+
+	malloc_mutex_lock(tsdn, &ecache->mtx);
```

Comment:
I see, you moved the lock here so extent_recycle_split now assumes it is locked. Should the function name make that obvious in addition to the assert?